### ISSUE TITLE
Improve Variable Names

### ISSUE THREAD

**Issue by tinymachine** [glyphic-co/textblock#21]
The current default `minLineHeight` is `1.33` and default `maxLineHeight` is `1.25`. Unusual to have a `min` greater than the `max`. Maybe a typo?

**Comment by theorosendorf** [comment]
It is confusing and probably requires elaboration.  In this case the min/max refers to the state of the width so you get more space the smaller the type, less for large. I’d love to merge in an edit if you wanted to make it. 

A related feature request is the ability to transpose the min/max values in the event someone would want to do that for some reason. As it is now only one way works.

**Comment by tinymachine** [comment]
Ah, I see -- I can understand the desire to transpose these, as sometimes longer line-lengths can benefit from greater line-height (leading) so the eye can more easily distinguish between lines when moving from the end of one line to the beginning of the next.

So min and max values for font-size, line-height, and grade are all dependent on container width, right? I'll call these former three attributes 'dependent attributes'. From an API perspective, what if you simply handled transposing based on whether the min or max value of these dependent attributes is larger? That is, if max > min, the relationship to container width is direct; if min > max, it's inverse.

I think the variable naming for `minFontSize` and `maxFontSize` (and all the other dependent attribute mins and maxes) might need to change to something like `minWidthFontSize` and `maxWidthFontSize` to make this clearer. A bit wordy, but much clearer, I think. (And this would also address my confusion about `minLineHeight` and `maxLineHeight`.)

**Comment by theorosendorf** [comment]
These sound great. Would it be smart to support both versions of the variable names for some time before phasing the old ones out?

**Comment by tinymachine** [comment]
I think supporting both sets of variable names would make sense for a short time, maybe with a deprecation warning in the console if the old names are being used? If following [semantic versioning](https://semver.org), maybe you'd only drop support for the previous variable names with a major release (or first public release)?

(Full disclosure: I have no experience releasing JS packages, so I don't know what best practices would be!)

**Comment by jessevdp** [comment]
> I think supporting both sets of variable names would make sense for a short time, maybe with a deprecation warning in the console if the old names are being used? If following semantic versioning, maybe you'd only drop support for the previous variable names with a major release (or first public release)?

You are absolutely correct. "Breaking Changes" are only allowed when releasing major versions. (1.x.x -> 2.0.0) And since this is a change to the public API of the software it is considered a breaking change.

It is indeed common to release the "new and improved" version side by side with the old solution. In this case a deprication notice is a good idea. (Just a console log if the old variable name is used.)

Deprication notices should also be placed in documentation; and it is a good idea to mention what the new solution would be so users can pick up on that right away.

---

Hope this helps!

**Comment by theorosendorf** [comment]
Hey @tinymachine, are you up to working on either of those two things (renaming the variables or adding ability to transpose the values)?

We have this old issues about [transposing the values](https://github.com/glyphic-co/textblock/issues/16). I'll reference this discussion there.

I created an [initial public release](https://github.com/glyphic-co/textblock/releases/tag/v0.9.9).

**Comment by tinymachine** [comment]
Hi @theorosendorf - sure, I'll give it a shot! (Might not get to it until the end of the week.) BTW thank you for sharing this project!

**Comment by theorosendorf** [comment]
My pleasure @tinymachine! I hope you enjoy using it as much as we have.

Hey @traviskochel, what's the protocol here? Should we add @tinymachine as a collaborator, or...?

**Comment by traviskochel** [comment]
@theorosendorf  Even if they're not a collaborator, people should be able to submit pull requests, and then you'll get an alert to review and merge it in. This is probably the standard approach.

Adding them as a collaborator will allow them to push changes directly to the repository without being reviewed. This would be useful if they're making lots of changes and you trust them.

Your call though. I'm fine with anything.

**Comment by tinymachine** [comment]
No need for collaborator status but thanks for the consideration, @theorosendorf!

**Comment by theorosendorf** [comment]
K, thanks @traviskochel, and thanks for the help @tinymachine!

**Comment by tinymachine** [comment]
Hi @theorosendorf and @traviskochel -- I'm making progress on updating the variable names and updating the `calc()` function to handle transposed min/max values. I want to make sure, though, that I'm respecting the relationship between the width-dependent attributes and the container's width.

Is the relationship between line-height and container width linear? It looks like font-size and variable grade are linearly related, but I'm struggling to parse the line-height (`calcleading`) calculations:

https://github.com/glyphic-co/textblock/blob/567a0e3354f8c02469f2dc7c5f7c9f041a341f3e/src/textblock.js#L45-L62

If it's linear, that's great. (Though it does seem at odds with this statement in the readme: "Textblock gives you the control you need for setting systems like modular scales".)

Thanks for any light you can shed on the line-height calculations!

**Comment by theorosendorf** [comment]
Yeah, it’s linear. And yes, a modular scale for a single type style in responsive terms could follow a non-linear scale (in effect, easing over a cubic bezier [a future-future enhancement that I have some ideas for]). But the term modular scale refers to the relation of size and space between static elements, i.e. the heading is X/a, the subhead is Y/b, the body Z/c.

**Comment by tinymachine** [comment]
Thanks for the clarification, @theorosendorf, and for the explanation about modular scale (yes, I was indeed conflating that with easing curves!).
